msteams-platform/bots/how-to/conversations/Workflow bot in Teams.md

```diff
@@ -0,0 +1,580 @@
+
+  1. `userIds`: A set of user MRIs for those who can trigger auto-refresh. For more information on how to add in `userIds` list in refresh section of Adaptive Card, see [fetch the roster or user profile](../get-teams-context.md#fetch-the-roster-or-user-profile)
+  1. `verb`: A string to identify the refresh action.
+  1. `data`: An optional data to associate with the refresh action.
+
+In the following sample a base card returns as command response that can auto-refresh to specific user, such as the command sender:
+
+```
+        import baseCard from "../adaptiveCards/baseCard.json";
+        import { AdaptiveCards } from "@microsoft/adaptivecards-tools";
+
+        export class MyCommandHandler1 implements TeamsFxBotCommandHandler {
+        triggerPatterns: TriggerPatterns = "helloWorld";
+
+        async handleCommandReceived(context: TurnContext, message: CommandMessage): 
+        Promise<string | Partial<Activity> | void> {
+        const refreshVerb = "userViewRefresh";        // verb to identify the refresh action
+        const userIds = [ context.activity.from.id ]; // users who will be refreshed
+        const data = { key: "value"};                 // optional data associated with the action
+
+        const responseCard = AdaptiveCards
+          .declare(baseCard)
+          .refresh(refreshVerb, userIds, data)
+          .render(cardData);
+    
+            return MessageFactory.attachment(CardFactory.adaptiveCard(responseCard));
+        }
+      }
+```
+
+* Second option enables user-specific view to refresh your Adaptive Card. This is a sample refresh action defined in `baseCard.json`:
+
+  ```baseCard.json
+
+  {
+    "type": "AdaptiveCard",
+    "refresh": {
+      "action": {
+        "type": "Action.Execute",
+        "title": "Refresh",
+        "verb": "userViewRefresh" ,
+        "data": {
+          "key": "value"
+        }
+      },
+      "userIds": [
+        "${userID}"
+      ]
+    },
+    "body": [
+      ...
+    ],
+    ...
+  }
+  ```
+
+You need to replace `${userID}` with user MRI in code, while rendering your card content.
+
+#### Add user-specific Adaptive Cards
+
+You need to design the user-specific Adaptive Card to refresh specific users such as  `responseCard.json` for `userA` in the sample. To get started, you can create a `responseCard.json` with the following content, and put it in `bot/src/adaptiveCards` folder:
+
+```responseCard.json
+
+{
+  "type": "AdaptiveCard",
+  "body": [
+    {
+      "type": "TextBlock",
+      "size": "Medium",
+      "weight": "Bolder",
+      "text": "This is a user-specific view"
+    }
+  ],
+  "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
+  "version": "1.4"
+}
+
+```
+
+#### Add card action handler to refresh views
+
+Add handler that implements `TeamsFxAdaptiveCardActionHandler` to process the refresh invoke activity that is automatically triggered in Teams.
+
+```JASON
```

Comment:
The `JASON` here is not correct, it's typescript code

Reply:
This has been corrected.